src/adapters/utils.py

```diff
@@ -180,6 +181,38 @@ def remote_file_exists(url):
     return r.status_code == 200
 
 
+# Copied from here: https://github.com/huggingface/huggingface_hub/blob/v0.25.0/src/huggingface_hub/file_download.py#L266
+def url_to_filename(url: str, etag: Optional[str] = None) -> str:
```

Comment:
should not add this function back again. Use hf_hub_download() instead.
https://github.com/huggingface/huggingface_hub/releases